### PR TITLE
Fix incorrect row index filter in CDCReader

### PR DESCRIPTION
**Pull request by xupefei**
## Description

This PR fixes a bug in CDCReader, where row index filters are incorrectly flipped for insert rows.

Bug description:
when a tuple `(remove, file, dv1)` is removed and re-added with a new DV, i.e., `(add, file, dv2)`, we expect that:

1. rows marked in `dv2` but not in `dv1` are deleted.
2. rows marked in `dv1` but not in `dv2` are inserted (during restore for example).

We diff two DVs to produce inline DVs and pass them to file scans.
The bug is in how we use the inline DVs -- we wrongly used the `IfContained` filter for Item (2), and therefore all rows that are not marked in DV are being returned. The correct filter type here is `IfNotContained`, which will return all marked rows.

## How was this patch tested?

New test.

### CODE REVIEW

**Comment by xupefei** [comment]
@cstavr @vkorukanti Could you review this PR?

File: core/src/main/scala/org/apache/spark/sql/delta/commands/cdc/CDCReader.scala [hunk 1]
```diff
@@ -773,7 +773,7 @@ trait CDCReaderImpl extends DeltaLogging {
             deltaLog.dataPath,
             snapshot,
             rowIndexFilters =
-              Some(fileActionsToRowIndexFilters(addFiles.toSeq, useIfContainedFilters = true))),
+              Some(fileActionsToIfNotContainedRowIndexFilters(addFiles.toSeq))),
```
Comment (cstavr) [inline core/src/main/scala/org/apache/spark/sql/delta/commands/cdc/CDCReader.scala:776]: nit: There is a typo inL763 and L782: generateFileActionWithInlineDv -> generateFileActionsWithInlineDv

File: core/src/main/scala/org/apache/spark/sql/delta/commands/cdc/CDCReader.scala [hunk 2]
```diff
@@ -973,16 +973,11 @@ trait CDCReaderImpl extends DeltaLogging {
   }
 
   /**
-   * Return a map of file paths to IfContained or IfNotContained row index filters, depending on
-   * the value of `indexFilterType`.
+   * Return a map of file paths to IfNotContained row index filters, to keep all marked rows.
```
Comment (cstavr) [inline core/src/main/scala/org/apache/spark/sql/delta/commands/cdc/CDCReader.scala:976]:
```suggestion
   * Return a map of file paths to IfNotContained row index filters, to keep only the marked rows.
```

File: core/src/test/scala/org/apache/spark/sql/delta/DeltaCDCSuite.scala [hunk 3]
```diff
@@ -963,6 +963,30 @@ class DeltaCDCScalaSuite extends DeltaCDCSuiteBase {
     }
   }
 
+  test("Repeated delete") {
+    withTempDir { dir =>
+      val path = dir.getAbsolutePath
+      val deltaLog = DeltaLog.forTable(spark, path)
+      spark.range(1, 10, 1, 1).write.format("delta").save(path)
+      sql(s"delete from delta.`$path` where id = 3")
+      sql(s"delete from delta.`$path` where id = 4")
```
Comment (cstavr) [inline core/src/test/scala/org/apache/spark/sql/delta/DeltaCDCSuite.scala:972]:
```suggestion
      sql(s"DELETE FROM delta.`$path` WHERE id = 3")
      sql(s"DELETE FROM delta.`$path` WHERE id = 4")
```

Comment (cstavr) [reply core/src/test/scala/org/apache/spark/sql/delta/DeltaCDCSuite.scala:972]: Use a file with 5 rows and include one more delete that will fully remove a file with a DV.

**Review by cstavr — COMMENTED** [review]
LGTM. Left some nits.

File: core/src/test/scala/org/apache/spark/sql/delta/DeltaCDCSuite.scala (hunk 3 above)
Comment (xupefei) [reply core/src/test/scala/org/apache/spark/sql/delta/DeltaCDCSuite.scala]: Done!

**Review by cstavr — APPROVED** [review]

**Review by vkorukanti — APPROVED** [review]

**Pull request closed without merging**